src/test/java/net/snowflake/client/jdbc/ResultSetStructuredTypesLatestIT.java

```diff
@@ -244,6 +308,44 @@ public void testReturnAsMap() throws SQLException {
         });
   }
 
+  @Test
+  public void testReturnAsMapWithNullableValues() throws SQLException {
+    SnowflakeObjectTypeFactories.register(SimpleClass.class, SimpleClass::new);
+    withFirstRow(
+        "select {'x':{'string':'one'},'y':null,'z':{'string':'three'}}::MAP(VARCHAR, OBJECT(string VARCHAR));",
+        (resultSet) -> {
+          Map<String, SimpleClass> map =
+              resultSet.unwrap(SnowflakeBaseResultSet.class).getMap(1, SimpleClass.class);
+          assertEquals("one", map.get("x").getString());
+          assertNull(map.get("y"));
+          assertEquals("three", map.get("z").getString());
+        });
+  }
+
+  @Test
+  public void testReturnNullAsObjectOfTypeMap() throws SQLException {
+    SnowflakeObjectTypeFactories.register(SimpleClass.class, SimpleClass::new);
+    withFirstRow(
+        "select null::MAP(VARCHAR, OBJECT(string VARCHAR));",
+        (resultSet) -> {
+          Map<String, String> map =
+              resultSet.unwrap(SnowflakeBaseResultSet.class).getObject(1, Map.class);
+          assertNull(map);
+        });
+  }
+
+  @Test
+  public void testReturnNullAsMap() throws SQLException {
+    SnowflakeObjectTypeFactories.register(SimpleClass.class, SimpleClass::new);
+    withFirstRow(
+        "select null::MAP(VARCHAR, OBJECT(string VARCHAR));",
+        (resultSet) -> {
+          Map<String, SimpleClass> map =
+              resultSet.unwrap(SnowflakeBaseResultSet.class).getMap(1, SimpleClass.class);
+          assertNull(map);
+        });
+  }
+
   @Test
   public void testReturnAsMapOfTimestampsNtz() throws SQLException {
     withFirstRow(
```

Comment:
Maybe I missed it, but I can't find any test related to `wasNull`? Also any tests for `readObject(Integer.class)` etc?

Reply:
It is indirectly tested by implementation in FewTypesSqlData.